bundles/org.openhab.binding.solarwatt/src/main/java/org/openhab/binding/solarwatt/internal/handler/EnergyManagerHandler.java

```diff
@@ -0,0 +1,397 @@
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.solarwatt.internal.channel.SolarwattChannelTypeProvider;
+import org.openhab.binding.solarwatt.internal.configuration.SolarwattBridgeConfiguration;
+import org.openhab.binding.solarwatt.internal.configuration.SolarwattThingConfiguration;
+import org.openhab.binding.solarwatt.internal.discovery.SolarwattDevicesDiscoveryService;
+import org.openhab.binding.solarwatt.internal.domain.SolarwattChannel;
+import org.openhab.binding.solarwatt.internal.domain.model.Device;
+import org.openhab.binding.solarwatt.internal.domain.model.EnergyManager;
+import org.openhab.binding.solarwatt.internal.exception.SolarwattConnectionException;
+import org.openhab.core.cache.ExpiringCache;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EnergyManagerHandler} is responsible for handling energy manager thing itself
+ * and handle data retrieval for the child things.
+ *
+ * @author Sven Carstens - Initial contribution
+ */
+@NonNullByDefault
+public class EnergyManagerHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EnergyManagerHandler.class);
+
+    private final EnergyManagerConnector connector;
+    private final SolarwattChannelTypeProvider channelTypeProvider;
+
+    private @Nullable ExpiringCache<Map<String, Device>> devicesCache;
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    private @Nullable ZoneId zoneId;
+
+    /**
+     * Guid of this energy manager itself.
+     */
+    private @Nullable String energyManagerGuid;
+
+    /**
+     * Runner for the {@link ExpiringCache} refresh.
+     *
+     * Triggers update of all child things.
+     */
+    private final Runnable refreshRunnable = () -> {
+        EnergyManagerHandler.this.updateChannels();
+        EnergyManagerHandler.this.updateAllChildThings();
+    };
+
+    /**
+     * Create the handler.
+     *
+     * @param thing for which the handler is responsible
+     * @param channelTypeProvider provider for the channels
+     * @param httpClient connect to energy manager via this client
+     */
+    public EnergyManagerHandler(final Bridge thing, final SolarwattChannelTypeProvider channelTypeProvider,
+            final HttpClient httpClient) {
+        super(thing);
+        this.logger.debug("{} created", this);
```

Comment:
Can this message be replaced by using the debugger? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging
```
void myFun() {
    logger.trace("Enter myfun"); // DONT, DONT, really DONT do that
    doSomething();
    logger.trace("Leave myfun"); // DONT, DONT, really DONT do that
}
```

Reply:
I removed all trace/debug logging.